src/main/java/org/apache/sling/api/redirect/RedirectResolver.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.apache.sling.api.redirect;
+
+
+import org.osgi.annotation.versioning.ProviderType;
+
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * <p>
+ * A RedirectResolver resolves a redirect in the context of a request. It resolves that redirect having first been bound
+ * to a Resource. This is typically achieved through the Adapt to mechanism eg resource.adaptTo(RedirectResolver.class).
+ * </p><p>
+ * Implementations should implement an AdapterFactory to perform the adaption to an implementation of this class. That class
```

Comment:
Not necessarily, a resource provider can implement this as well; I think it makes sense to mention this here as well

Reply:
done